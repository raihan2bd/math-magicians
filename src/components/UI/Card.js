import React from 'react';
import PropTypes from 'prop-types';

import classes from './Card.module.css';

const Card = (props) => {
  const { extraclass, children } = props;

  return (
    <div
      className={extraclass ? `${classes.card} ${extraclass}` : classes.card}
    >
      {children}
    </div>
  );
};

Card.propTypes = {
  extraclass: PropTypes.string,
  children: PropTypes.node.isRequired,
};

Card.defaultProps = {
  extraclass: '',
};

export default Card;
