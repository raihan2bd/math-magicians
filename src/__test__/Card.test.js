import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Card from '../components/UI/Card';

describe('Test Card component with snapshot', () => {
  test('should match with snapshot', () => {
    const card = renderer
      .create(
        <Card>
          <h1>Hello from Card Component</h1>
        </Card>,
      )
      .toJSON();

    expect(card).toMatchSnapshot();
  });

  test('should be in the document', () => {
    render(
      <Card>
        <h1>Hello from Card Component</h1>
      </Card>,
    );

    expect(screen.getByText('Hello from Card Component')).toBeInTheDocument();
  });
});
