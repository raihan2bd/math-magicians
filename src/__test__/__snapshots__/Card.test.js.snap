// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Card component with snapshot should match with snapshot 1`] = `
<div
  className="card"
>
  <h1>
    Hello from Card Component
  </h1>
</div>
`;
